### ISSUE TITLE
Mocking Carbon with setTestNow() returns incorrect results when specifying certain timezones

### ISSUE THREAD

**Issue by PHLAK** [briannesbitt/Carbon#999]
The following fails:

```php
php > Carbon\Carbon::setTestNow(Carbon\Carbon::create(2017, 7, 1, 0, 0, 0, 'America/Phoenix'));

php > var_dump(Carbon\Carbon::now());
object(Carbon\Carbon)#2 (3) {
  ["date"]=>
  string(26) "2017-07-01 00:00:00.000000"
  ["timezone_type"]=>
  int(3)
  ["timezone"]=>
  string(15) "America/Phoenix"
}

php > var_dump(new Carbon\Carbon('first friday of this month', 'America/Phoenix'));
object(Carbon\Carbon)#2 (3) {
  ["date"]=>
  string(26) "2017-07-06 17:00:00.000000"
  ["timezone_type"]=>
  int(3)
  ["timezone"]=>
  string(15) "America/Phoenix"
}
```

The first Friday of July 2017 should be `2017-07-07 00:00:00.000000` but the returned object shows `2017-07-06 17:00:00.000000` which happens to be -7 hours from the expected time. It might be a coincidence but this matches the `America/Phoenix` timezone offset of -7 hours UTC.

Here's an example omitting the timezone:

```php
php > Carbon\Carbon::setTestNow(Carbon\Carbon::create(2017, 7, 1, 0, 0, 0));

php > var_dump(Carbon\Carbon::now());
object(Carbon\Carbon)#3 (3) {
  ["date"]=>
  string(26) "2017-07-01 00:00:00.000000"
  ["timezone_type"]=>
  int(3)
  ["timezone"]=>
  string(14) "America/Denver"
}

php > var_dump(new Carbon\Carbon('first friday of this month'));
object(Carbon\Carbon)#3 (3) {
  ["date"]=>
  string(26) "2017-07-07 00:00:00.000000"
  ["timezone_type"]=>
  int(3)
  ["timezone"]=>
  string(14) "America/Denver"
}
```

The resulting object is correct.

**Comment by PHLAK** [comment]
Here's some additional tests with the `America/Los_Angeles` and `America/New_York` timezones, both returning incorrect results.

```php
php > Carbon\Carbon::setTestNow(Carbon\Carbon::create(2017, 7, 1, 0, 0, 0, 'America/Los_Angeles'));

php > var_dump(Carbon\Carbon::now());

object(Carbon\Carbon)#2 (3) {
  ["date"]=>
  string(26) "2017-07-01 00:00:00.000000"
  ["timezone_type"]=>
  int(3)
  ["timezone"]=>
  string(19) "America/Los_Angeles"
}

php > var_dump(new Carbon\Carbon('first friday of this month', 'America/Los_Angeles'));
object(Carbon\Carbon)#2 (3) {
  ["date"]=>
  string(26) "2017-07-06 17:00:00.000000"
  ["timezone_type"]=>
  int(3)
  ["timezone"]=>
  string(19) "America/Los_Angeles"
}
```

```php
php > Carbon\Carbon::setTestNow(Carbon\Carbon::create(2017, 7, 1, 0, 0, 0, 'America/New_York'));

php > var_dump(Carbon\Carbon::now());
object(Carbon\Carbon)#3 (3) {
  ["date"]=>
  string(26) "2017-07-01 00:00:00.000000"
  ["timezone_type"]=>
  int(3)
  ["timezone"]=>
  string(16) "America/New_York"
}

php > var_dump(new Carbon\Carbon('first friday of this month', 'America/New_York'));
object(Carbon\Carbon)#3 (3) {
  ["date"]=>
  string(26) "2017-07-06 20:00:00.000000"
  ["timezone_type"]=>
  int(3)
  ["timezone"]=>
  string(16) "America/New_York"
}
```

**Comment by PHLAK** [comment]
I set my `date.timezone` to `America/Phoenix` in `php.ini` and now it's working correctly for the `America/Phoenix` timezone but other timezones (including `America/Denver` now) still exhibit the symptoms.

**Comment by kylekatarnls** [comment]
Hi, it could rely to our modify fix so it should be fixed now with 1.23 version. If not, please tell me, I'll re-open.
